.github/workflows/leapfrog-py-test.yaml

```diff
@@ -0,0 +1,87 @@
+name: Test
+
+on:
+  pull_request:
+    branches:
+      - main
+      - python-interface-poc
+  push:
+    branches:
+      - main
+      - python-interface-poc
+
+jobs:
+  run:
+
+    runs-on: ${{ matrix.config.os }}
+
+    name: ${{matrix.config.os }} (${{ matrix.config.py }})
+
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - {os: macos-latest,   py: '3.12'}
+          - {os: ubuntu-latest,  py: '3.8'}
+          - {os: ubuntu-latest,  py: '3.10'}
+          - {os: ubuntu-latest,  py: '3.11'}
+          - {os: ubuntu-latest,  py: '3.12'}
+          - {os: windows-latest, py: '3.12'}
```

Comment:
why not 3.9?

Reply:
Ah just missed that when I went through and updated the versions. Testing 3.9 - 3.12 which I think should be enough.